test/job/concurrentreads/concurrentreads.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2021 Andy Dalton
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"strings"
+	"sync"
+
+	"github.com/adalton/teleport-exercise/pkg/jobmanager"
+)
+
+func runTest() {
+
+	numValues := 100 * 1000
+
+	job := jobmanager.NewJob("theOwner", "my-test", nil,
+		"/bin/bash",
+		"-c",
+		"for ((i = 0; i < 100; ++i)); do for((j = 0; j < 1000; ++j)); do echo $RANDOM; done; sleep 0.25; done",
+	)
+
+	if err := job.Start(); err != nil {
+		panic(err)
+	}
+
+	const numGoroutines = 100
+	var buckets [numGoroutines][]byte
```

Comment:
You could start with a `var buckets [numGoroutines]bytes.Buffer` and skip the transformation below.

Reply:
Ah, nice -- thank you for the pointer!